Source/Core/DolphinQt/Debugger/JitBlockTableModel.cpp

```diff
@@ -0,0 +1,453 @@
+  case Column::HostFarCodeSize:
+    return static_cast<qulonglong>(jit_block.far_end - jit_block.far_begin);
+  }
+  const JitBlock::ProfileData* const profile_data = jit_block.profile_data.get();
+  if (profile_data == nullptr)
+    return QVariant();
+  switch (column)
+  {
+  case Column::RunCount:
+    return static_cast<qulonglong>(profile_data->run_count);
+  case Column::CyclesSpent:
+  case Column::CyclesPercent:
+    return static_cast<qulonglong>(profile_data->cycles_spent);
+  case Column::CyclesAverage:
+    if (profile_data->run_count == 0)
+      return QVariant();
+    return static_cast<double>(profile_data->cycles_spent) / profile_data->run_count;
+  case Column::TimeSpent:
+  case Column::TimePercent:
+    return static_cast<qulonglong>(profile_data->time_spent.count());
+  case Column::TimeAverage:
+    if (profile_data->run_count == 0)
+      return QVariant();
+    return static_cast<double>(profile_data->time_spent.count()) / profile_data->run_count;
+  }
+  static_assert(Column::NumberOfColumns == 14);
+  Common::Unreachable();
+}
+
+const JitBlock& JitBlockTableModel::GetJitBlock(const QModelIndex& index) const
+{
+  ASSERT(index.isValid());
+  return m_jit_blocks[index.row()];
+}
+
+void JitBlockTableModel::OnJitCacheCleared()
+{
+  Update(Core::GetState(m_system));
+}
+
+void JitBlockTableModel::OnJitProfileDataWiped()
+{
+  UpdateProfileData();
+}
+
+void JitBlockTableModel::OnUpdateDisasmDialog()
+{
+  // This should hopefully catch all the little things that lead to stale JitBlock references.
+  Update(Core::GetState(m_system));
+}
+
+void JitBlockTableModel::OnPPCSymbolsUpdated()
+{
+  UpdateSymbols();
+}
+
+void JitBlockTableModel::OnPPCBreakpointsChanged()
+{
+  Update(Core::GetState(m_system));
+}
+
+void JitBlockTableModel::OnEmulationStateChanged(Core::State state)
+{
+  Update(state);
+}
+
+void JitBlockTableModel::OnShowSignal()
+{
+  Show();
+}
+
+void JitBlockTableModel::OnHideSignal()
+{
+  Hide();
+}
+
+void JitBlockTableModel::OnSortIndicatorChanged(int logicalIndex, Qt::SortOrder)
+{
+  m_sorting_by_symbols = logicalIndex == Column::Symbol;
+}
+
+void JitBlockTableModel::OnFilterSymbolTextChanged(const QString& string)
+{
+  m_filtering_by_symbols = !string.isEmpty();
+}
+
+QVariant JitBlockTableModel::data(const QModelIndex& index, int role) const
+{
+  switch (role)
+  {
+  case Qt::DisplayRole:
+    return DisplayRoleData(index);
+  case Qt::TextAlignmentRole:
+    return TextAlignmentRoleData(index);
+  case UserRole::SortRole:
+    return SortRoleData(index);
+  }
+  return QVariant();
+}
+
+QVariant JitBlockTableModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+  if (orientation != Qt::Horizontal || role != Qt::DisplayRole)
+    return QVariant();
+
+  // i18n: These abbreviated table header display names have unabbreviated copies in JITWidget.cpp.
+  // Try to match the translations while keeping these ones short so they will fit in the columns.
```

Comment:
This won't work. An 18n comment has to be immediately above the string you want to attach it to.

If you want translators to see this, the way to do it would be to make a copy of it for each of the strings here, which might be overkill.